### DISCUSSION TITLE
error: `std::ranges` has not be defined

### DISCUSSION THREAD

**Discussion by MustacheMan2** [DarkflameUniverse/DarkflameServer#1441] [Q&A] [answered]
Whenever it gets to building RenderComponent.cpp, this happens.
![image](https://github.com/DarkflameUniverse/DarkflameServer/assets/48499798/1dca5543-a03a-4aae-8518-3c66930f93df)

Tested locally using WSL and it was able to build. If this answers anything, this server has been around since day one of DLU's launch, so it could be something wrong with the db or some outdated dependencies, though everything seems up to date. Happens on a fresh clone as well. Posting here first in case it's specific to this machine.

**Comment by MustacheMan2** [comment]
An update to 22.04 resolves this.

**Comment by aronwk-aaron** [comment, marked as answer]
gcc/g++ 11 is now required, this can be resolved by upgrading to 22.04 on ubuntu or just upgrading g++/gcc

**Reply by aronwk-aaron** [reply]
https://webhostinggeeks.com/howto/how-to-upgrade-gcc-on-ubuntu/
example of upgrading
